src/cache.rs

```diff
@@ -23,7 +23,7 @@ type DB = DBWithThreadMode<MultiThreaded>;
 ///
 /// path is the path of the rocks db database
 pub(crate) struct Cache {
-    db: DB,
+    pub db: DB,
```

Comment:
This smells a little to me, could you elaborate on why it's necessary? It will make it more difficult if we want to change DB implementation in the future.

Reply:
It's used by the client::sync to retrieve current live note's nullifiers.
In order to not compute the nullifier from the note, the key is required to be iterated.

However I agree with your concern. We can achieve the same result adding a method that return the note's keys (or an iterator with key-value -> (nullifier-note)

Reply:
Up to you, as long as this guy the db is encapsulated it sounds good to me!